Flaky (but deterministic) test: DockerTrustSuite.TestTrustedIsolated(Create|Pull)

The 1st run succeeds, but the 2nd and 3rd one always fail.

```console
$ TESTFLAGS="-test.count 3 -check.f DockerTrustSuite.TestTrustedIsolatedCreate" make test-integration-cli
...
INFO: Testing against a local daemon
PASS: docker_cli_create_test.go:326: DockerTrustSuite.TestTrustedIsolatedCreate 0.417s
OK: 1 passed

----------------------------------------------------------------------
FAIL: docker_cli_create_test.go:326: DockerTrustSuite.TestTrustedIsolatedCreate

docker_cli_create_test.go:330:
    // Try create
    icmd.RunCmd(icmd.Command(dockerBinary, "--config", "/tmp/docker-isolated-create", "create", repoName), trustedCmd).Assert(c, SuccessTagging)
/go/src/github.com/docker/docker/pkg/testutil/cmd/command.go:64:
    t.Fatalf("at %s:%d\n%s", filepath.Base(file), line, err.Error())
... Error: at docker_cli_create_test.go:330

Command: /go/src/github.com/docker/docker/bundles/1.14.0-dev/dynbinary-client/docker --config /tmp/docker-isolated-create create 127.0.0.1:5000/dockercli/trusted-isolated-create:latest
ExitCode: 1, Error: exit status 1
Stdout: 
Stderr: could not rotate trust to a new trusted root: failed to validate data with current trusted certificates


Failures:
ExitCode was 1 expected 0
Expected stdout to contain "Tagging"
Expected no error


OOPS: 0 passed, 1 FAILED
--- FAIL: Test (0.60s)

----------------------------------------------------------------------
FAIL: docker_cli_create_test.go:326: DockerTrustSuite.TestTrustedIsolatedCreate

docker_cli_create_test.go:330:
    // Try create
    icmd.RunCmd(icmd.Command(dockerBinary, "--config", "/tmp/docker-isolated-create", "create", repoName), trustedCmd).Assert(c, SuccessTagging)
/go/src/github.com/docker/docker/pkg/testutil/cmd/command.go:64:
    t.Fatalf("at %s:%d\n%s", filepath.Base(file), line, err.Error())
... Error: at docker_cli_create_test.go:330

Command: /go/src/github.com/docker/docker/bundles/1.14.0-dev/dynbinary-client/docker --config /tmp/docker-isolated-create create 127.0.0.1:5000/dockercli/trusted-isolated-create:latest
ExitCode: 1, Error: exit status 1
Stdout: 
Stderr: could not rotate trust to a new trusted root: failed to validate data with current trusted certificates


Failures:
ExitCode was 1 expected 0
Expected stdout to contain "Tagging"
Expected no error


OOPS: 0 passed, 1 FAILED
--- FAIL: Test (0.60s)
FAIL
...
```

version: master (a7c3389a8223449ed72d796b33a667e603811db8)

The error seems from https://github.com/docker/notary/blob/master/trustpinning/certs.go#L134

This issue is not critical for production, but makes it hard to debug other flaky tests with `TESTFLAGS=-test.count %d`.

--- a/integration-cli/docker_cli_create_test.go
+++ b/integration-cli/docker_cli_create_test.go
@@ -321,6 +321,7 @@ func (s *DockerTrustSuite) TestTrustedIsolatedCreate(c *check.C) {
 
 	// Try create
 	icmd.RunCmd(icmd.Command(dockerBinary, "--config", "/tmp/docker-isolated-create", "create", repoName), trustedCmd).Assert(c, SuccessTagging)
+	defer os.RemoveAll("/tmp/docker-isolated-create")
 
 	dockerCmd(c, "rmi", repoName)
 }

